src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -37,23 +34,17 @@ const TopicFragment = ({
   searchText,
   withoutRecent = false
 }) => {
-  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
   const {followTopic} = useChatClientHook();
+  const {
+    topics,
+    updateFollowTopicDiscoveryContext,
+    topicExchangeFollower: exhangeFollower
+  } = useDiscovery();
 
   const navigation = useNavigation();
 
-  const [myId, setMyId] = React.useState('');
-
   const isReady = useIsReady();
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here seem to be an improvement in terms of code readability and maintainability. The use of context for state management has been replaced with a custom hook `useDiscovery`, which is a more modular approach. However, it's important to ensure that the `useDiscovery` hook is implemented correctly and efficiently. Also, make sure that the `updateFollowTopicDiscoveryContext` function handles all edge cases and updates the state as expected.



<!-- This is an auto-generated comment by OSS CodeRabbit -->